lib/spack/spack/util/environment.py

```diff
@@ -874,9 +874,17 @@ def from_sourcing_file(
             ]
         )
 
+        # Implements a no op during forwards operation on respective os
         # Compute the environments before and after sourcing
+
+        before_kwargs = {**kwargs}
+        if sys.platform == "win32":
+            # Windows cannot call nul (os.devnull), only pipe output to it
+            # so we call echo on our null fd and then pipe it to null
```

Comment:
IMO the following is clearer

> Windows cannot source os.devnull, but it can echo it, so we override the "source" action in the method that extracts the env (environment_after_sourcing_files)

* "call" -> "source"
* "only pipe output to it": we are echoing `os.devnull`, so we are doing more than directing output to it

Reply:
Thanks! Done!